src/main/webapp/assets/modules/post-contest/postContestController.js

```diff
@@ -217,17 +175,86 @@ techlooper.controller("postContestController", function ($scope, $http, jsonValu
         lastStep = true;
       }
     }
-  }
+  };
 
   $scope.config = {
     reviewStyle: resourcesService.reviewStyleConfig,
     qualityIdea: resourcesService.qualityIdeaConfig
-  }
+  };
 
   $scope.changeState(state.challenge);
 
   $scope.detectDeleteKey = function (e) {
-    if (e.which == 8)
-      return false;
+    if (e.which == 8) return false;
   };
+
+  $scope.contest = {};
+
+  var gtDate = function (modelValue, contestProperty) {
+    if (!modelValue) return true;
+
+    var currentDate = moment(modelValue, jsonValue.dateFormat);
+    var compareDate = moment($scope.contest[contestProperty], jsonValue.dateFormat);
+    if (!compareDate.isValid()) return false;
+    return currentDate.isAfter(compareDate, 'day');
+  };
+
+  // timeline validators
+  $scope.$watch(function () {
+    return [
+      $scope.contest.startDate, $scope.contest.registrationDate,
+      $scope.contest.ideaSubmissionDate, $scope.contest.uxSubmissionDate,
+      $scope.contest.prototypeSubmissionDate, $scope.contest.submissionDate,
+      $scope.ideaChecked, $scope.uiuxChecked, $scope.prototypeChecked
+    ];
+  }, function () {
+
+    var currentDate = undefined;
+    var compareDate = undefined;
+    var valid = undefined;
+
+    //start-date
+    currentDate =  moment($scope.contest.startDate, jsonValue.dateFormat);
+    if (currentDate.isValid()) {
+      compareDate = moment().add(4, 'weeks');
+      valid = currentDate.isBetween(moment(), compareDate, 'day') || currentDate.isSame(moment(), "day");
+      $scope.timelineForm.startDate.$setValidity("in4w", valid);
+    }
+
+    //registration-date
+    valid = gtDate($scope.contest.registrationDate, "startDate");
+    $scope.timelineForm && $scope.timelineForm.registrationDate.$setValidity("gtStartDate", valid);
```

Comment:
Line is too long.<br>Expected an assignment or function call and instead saw an expression.